### PR TITLE
Basic auth UI

### PR DESCRIPTION
**Pull request by Aierie**
Basic auth UI + tests to ensure that correct UI is shown in response to various states of services, and events trigger the correct callbacks. Styling improvements to come in a separate PR.

The auth steps in order are `WALLET_CONNECT` -> `HUB_AUTH` -> `DONE`, with a `LOADING` state to prevent weird transitions at first load, and a fallback state for any unknown errors.

This is to provide an auth flow that brings the user through connecting to layer 2 and authenticating with hub, in a way that is usable on mobile. The component (and services it depends on) may be moved or copied to SSR web depending on the auth method we settle on.

### CODE REVIEW

File: packages/web-client/app/components/auth/index.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import HubAuthentication from '@cardstack/web-client/services/hub-authentication';
+import UA from '@cardstack/web-client/services/ua';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { race, rawTimeout, task, TaskGenerator } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import config from '@cardstack/web-client/config/environment';
+
+const AUTH_STEPS = {
```
Comment (jurgenwerk) [inline packages/web-client/app/components/auth/index.ts:12]: The other day I came across this article https://swizec.com/blog/stop-shouting-shouting/ and it struck a chord with me 😀 I know we use this pattern across the app but wondering what you think about simply using a type for this

**Review by jurgenwerk — APPROVED** [review]
Great, looks like a solid start for the mobile auth.

File: packages/web-client/app/components/auth/index.ts (hunk 1 above)
Comment (Aierie) [reply packages/web-client/app/components/auth/index.ts:12]: i think it's ok, but we have no types in templates.... so I'm shouting to get the point across to hbs

Comment (Aierie) [reply packages/web-client/app/components/auth/index.ts:12]: If templates had types I would be happy to use types

**Pull request merged**